test-apps/imodel-transformer/src/test/Transformer.test.ts

```diff
@@ -84,4 +84,20 @@ describe("imodel-transformer", () => {
     assert.isAtLeast(count(sourceDb, PhysicalPartition.classFullName), 2);
     targetDb.close();
   });
+
+  it("should exclude categories", async () => {
+    const targetDbFileName = initOutputFile("CompatibilityTestSeed-Simplified.bim");
+    const targetDb = SnapshotDb.createEmpty(targetDbFileName, {
+      rootSubject: { name: `${sourceDb.rootSubject.name}-CategoryExcluded` },
+      ecefLocation: sourceDb.ecefLocation,
+    });
+
+    await Transformer.transformAll(new BackendRequestContext(), sourceDb, targetDb, { excludeCategories: ["TestSpatialCategory"] });
+    const numSourceElements = count(sourceDb, GeometricElement.classFullName);
+    assert.isAtLeast(numSourceElements, 24);
+    const numTargetElems = count(targetDb, GeometricElement.classFullName);
+    assert.isBelow(numTargetElems, numSourceElements);
+    assert.isAtLeast(numTargetElems, 18);

```

Comment:
Not really testing that anything was excluded. Should check that the Category exists in the source but not the target and Elements on that Category exist in the source but not the target.

Reply:
the count tests are supposedly a way of testing that some geometric elements were pruned but I'll do some much better ecsql checks to ensure no geometric elements with that category exist

Reply:
My point was that 24 isAtLeast 18. You'd have to use a different assertion to detect that something was actually excluded. The more explicit checks are better though...

Reply:
the `isBelow` tests that they are not both 24. But either way I agree I'll switch to explict checks on the affected elements

Reply:
let me know if you think I missed something in the new tests